pkg/cloudprovider/drift.go

```diff
@@ -25,16 +25,26 @@ import (
 	"github.com/aws/karpenter-core/pkg/cloudprovider"
 	"github.com/aws/karpenter/pkg/apis/v1alpha1"
 	"github.com/aws/karpenter/pkg/providers/amifamily"
+	"github.com/aws/karpenter/pkg/providers/instance"
 	"github.com/aws/karpenter/pkg/utils"
 )
 
 func (c *CloudProvider) isNodeTemplateDrifted(ctx context.Context, machine *v1alpha5.Machine, provisioner *v1alpha5.Provisioner, nodeTemplate *v1alpha1.AWSNodeTemplate) (bool, error) {
+	ec2Instance, err := c.getInstance(ctx, machine.Status.ProviderID)
+	if err != nil {
+		return false, cloudprovider.IgnoreMachineNotFoundError(err)
```

Comment:
Would be nice if we could do the `IgnoreMachineNotFoundError()` call in the parent call and just return err here and in the other places, since it seems like we always want to ignore not found